File: examples/src/hooks.rs
//! Hook implementations for Nested Command Demo
//! 
//! This file contains the business logic for your CLI commands.
//! Implement the hook functions below to handle your CLI commands.
//! 
//! Each command in your CLI corresponds to a hook function.

use clap::ArgMatches;
use anyhow::{Result, anyhow};
/// Hook function for 'simple' command
/// Simple command that works today
pub fn on_simple(matches: &ArgMatches) -> Result<()> {
    // TODO: Implement your business logic here
    println!("Executing simple command...");
    
    // Extract arguments
if let Some(message) = matches.get_one::<String>("message") {
        println!("message: {}", message);
    }
    
    // Extract options
let verbose = matches.get_flag("verbose");
    println!("verbose: {}", verbose);
    
    // Return Ok(()) for success, Err(...) for error
    Ok(())
}
/// Hook function for 'database' command
/// Database operations
pub fn on_database(matches: &ArgMatches) -> Result<()> {
    // TODO: Implement your business logic here
    println!("Executing database command...");
    
    // Return Ok(()) for success, Err(...) for error
    Ok(())
}
/// Hook function for 'api' command
/// API management
pub fn on_api(matches: &ArgMatches) -> Result<()> {
    // TODO: Implement your business logic here
    println!("Executing api command...");
    
    // Return Ok(()) for success, Err(...) for error
    Ok(())
}
/// Simple hook manager for compatibility with other templates
pub struct HookManager;

impl HookManager {
    /// Create a new hook manager
    pub fn new() -> Self {
        Self
    }
    
    /// Execute a hook by name
    pub fn execute_hook(&self, name: &str, matches: &ArgMatches) -> Result<()> {
        match name {            "on_simple" => on_simple(matches),            "on_database" => on_database(matches),            "on_api" => on_api(matches),            _ => Err(anyhow::anyhow!("Unknown hook: {}", name)),
        }
    }
}

impl Default for HookManager {
    fn default() -> Self {
        Self::new()
    }
}

// Add any utility functions or structures here

File: src/examples/validation/test_universal_typescript_output/src/hooks.ts
/**
 * Hook implementations for Test Universal CLI
 * 
 * This file contains the business logic for your CLI commands.
 * Implement the hook functions below to handle your CLI commands.
 * 
 * IMPORTANT: Hook names must use snake_case with 'on_' prefix
 * Example:
 * - Command 'hello' -> Hook function 'on_hello'
 * - Command 'hello-world' -> Hook function 'on_hello_world'
 */

// Import any modules you need here
import * as fs from 'fs';
import * as path from 'path';
/**
 * Say hello to someone * @param name Name to greet * @param options Command options * @param options.greeting Custom greeting message * @returns Promise<number> - Exit code (0 for success)
 */
export async function on_hello(    name: string,    options: {        greeting?: string;    }): Promise<number> {
    // Add your business logic here
    console.log('Hook on_hello called');    console.log('Arguments:', { name });    console.log('Options:', options);    
    // Return 0 for success, non-zero for error
    return 0;
}